llvm/test/Instrumentation/AddressSanitizer/AMDGPU/asan-dynamic-lds-test.ll

```diff
@@ -0,0 +1,231 @@
+; CHECK-NEXT:    [[TMP65:%.*]] = call ptr @llvm.returnaddress(i32 0)
+; CHECK-NEXT:    [[TMP66:%.*]] = ptrtoint ptr [[TMP65]] to i64
+; CHECK-NEXT:    [[TMP67:%.*]] = ptrtoint ptr addrspace(1) [[TMP64]] to i64
+; CHECK-NEXT:    call void @__asan_free_impl(i64 [[TMP67]], i64 [[TMP66]])
+; CHECK-NEXT:    br label %[[END]]
+; CHECK:       [[END]]:
+; CHECK-NEXT:    ret void
+;
+WId:
+  %0 = call i32 @llvm.amdgcn.workitem.id.x()
+  %1 = call i32 @llvm.amdgcn.workitem.id.y()
+  %2 = call i32 @llvm.amdgcn.workitem.id.z()
+  %3 = or i32 %0, %1
+  %4 = or i32 %3, %2
+  %5 = icmp eq i32 %4, 0
+  br i1 %5, label %Malloc, label %21
+
+Malloc:                                           ; preds = %WId
+  %6 = load i32, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, align 4
+  %7 = load i32, ptr addrspace(1) getelementptr inbounds (%llvm.amdgcn.sw.lds.k0.md.type, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, i32 0, i32 0, i32 2), align 4
+  %8 = add i32 %6, %7
+  %9 = call ptr addrspace(4) @llvm.amdgcn.implicitarg.ptr()
+  %10 = getelementptr inbounds ptr addrspace(4), ptr addrspace(4) %9, i64 15
+  store i32 %8, ptr addrspace(1) getelementptr inbounds (%llvm.amdgcn.sw.lds.k0.md.type, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, i32 0, i32 1, i32 0), align 4
+  %11 = load i32, ptr addrspace(4) %10, align 4
+  store i32 %11, ptr addrspace(1) getelementptr inbounds (%llvm.amdgcn.sw.lds.k0.md.type, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, i32 0, i32 1, i32 1), align 4
+  %12 = add i32 %11, 0
+  %13 = udiv i32 %12, 1
+  %14 = mul i32 %13, 1
+  store i32 %14, ptr addrspace(1) getelementptr inbounds (%llvm.amdgcn.sw.lds.k0.md.type, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, i32 0, i32 1, i32 2), align 4
+  %15 = add i32 %8, %14
+  %16 = zext i32 %15 to i64
+  %17 = call ptr @llvm.returnaddress(i32 0)
+  %18 = ptrtoint ptr %17 to i64
+  %19 = call i64 @__asan_malloc_impl(i64 %16, i64 %18)
+  %20 = inttoptr i64 %19 to ptr addrspace(1)
+  store ptr addrspace(1) %20, ptr addrspace(3) @llvm.amdgcn.sw.lds.k0, align 8
+  br label %21
+
+21:                                               ; preds = %Malloc, %WId
+  %xyzCond = phi i1 [ false, %WId ], [ true, %Malloc ]
+  call void @llvm.amdgcn.s.barrier()
+  %22 = load i32, ptr addrspace(1) getelementptr inbounds (%llvm.amdgcn.sw.lds.k0.md.type, ptr addrspace(1) @llvm.amdgcn.sw.lds.k0.md, i32 0, i32 1, i32 0), align 4
+  %23 = getelementptr inbounds i8, ptr addrspace(3) @llvm.amdgcn.sw.lds.k0, i32 %22
+  call void @llvm.donothing() [ "ExplicitUse"(ptr addrspace(3) @llvm.amdgcn.k0.dynlds) ]
+  store i8 7, ptr addrspace(3) %23, align 4
+  br label %CondFree
+
+CondFree:                                         ; preds = %21
+  call void @llvm.amdgcn.s.barrier()
+  br i1 %xyzCond, label %Free, label %End
+
+Free:                                             ; preds = %CondFree
+  %24 = load ptr addrspace(1), ptr addrspace(3) @llvm.amdgcn.sw.lds.k0, align 8
```

Comment:
Use named values in tests

Reply:
The test IR input is generated by running amdgpu-sw-lower-lds pass. Unnamed values are seen because of this. Will update them with named value if needed.